public/js/p3/widget/CircularViewerContainer.js

```diff
@@ -123,6 +124,95 @@ define([
 			}));
 		},
 
+		addSpGeneTrack: function(title, title_tooltip, gid, filter, strand, fill, stroke, background){
+			var fields = ["feature_id", "feature_type", "sequence_id", "segments", "gi", "na_length", "pos_group", "strand", "public", "aa_length", "patric_id", "owner",
+				"location", "protein_id", "refseq_locus_tag", "taxon_id", "accession", "end", "genome_name", "product", "genome_id", "annotation", "start"];
+				
+			var sp_fields = ["feature_id","patric_id", "evidence", "property", "source"]
+
+			//var sp_query = "?eq(genome_id," + this.state.genome_ids[0] + ")&in(property,(%22Antibiotic%20Resistance%22,%22Virulence%20Factor%22,%22Transporter%22,%22Essential%20Gene%22))&sort(+patric_id,+property)" + "&select(" + sp_fields.join(",") + ")&limit(25000)";
+			var sp_query = "?eq(genome_id," + this.state.genome_ids[0] + ")" + filter + "&sort(+patric_id,+property)" + "&select(" + sp_fields.join(",") + ")&limit(25000)";
```

Comment:
this should work, but a little bit inefficient. 

1. you will not need anything more than feature_id in selection. that is what you're using in line 176
2. do you need to sort? i guess your filter already narrow down one property at a time.

Reply:
This was done before deciding only one property per track. The original plan was to have multiple proteries on one track.

Reply:
so, can we clean up then?

Reply:
clean up and committed the changes